src/Extensions/Parser.hs

```diff
@@ -111,4 +111,15 @@ cppP = [] <$ many newline <* try (char '#' <* noneOf "-") <* manyTill anyChar (t
 newLines :: Parser ()
 newLines = () <$ many (space <|> endOfLine)
 
+{- | Parse 'Extension' from a string. 'Read' instance for 'Extension'
+doesn't always work since some extensions are named differently.
+-}
+readExtension :: String -> Maybe Extension
+readExtension = \case
+    "GeneralisedNewtypeDeriving" -> Just GeneralizedNewtypeDeriving
+    "NamedFieldPuns" -> Just RecordPuns
+    "Rank2Types" -> Just RankNTypes
+    "CPP" -> Just Cpp
```

Comment:
Seems like `Cpp` is not working in module files.
Sorry, I am a bit confused in all these `Extension`s data types and their allowed scope (where they would work).

Reply:
Yes, this is indeed confusing... Cabal has its own type for extensions which is not the same as GHC extension type. Cabal works only with `.cabal` files and parses extensions from this file. And GHC only works with Haskell modules, it doesn't know anything about `.cabal` files. Problems happen when an extension is written in one way in a file (like `CPP`) but constructor name is different (`Cpp`), so `read` instance won't work. But this means that I also need to handle `Cpp` string as `Nothing`...